fcrepo-kernel-modeshape/src/test/java/org/fcrepo/integration/kernel/modeshape/FedoraResourceImplIT.java

```diff
@@ -185,15 +186,40 @@ public void testRandomNodeGraph() {
 
     @Test
     public void testLastModified() throws RepositoryException {
-        final String pid = UUID.randomUUID().toString();
+        final String pid = getRandomPid();
+        containerService.findOrCreate(session, "/" + pid);
+
+        session.save();
+        session.logout();
+        session = repo.login();
+
+        final Container obj2 = containerService.findOrCreate(session, "/" + pid);
+        final Date created = roundDate(obj2.getCreatedDate());
+        final Date modified = roundDate(obj2.getLastModifiedDate());
+        assertFalse(modified + " should not be before " + created, modified.before(created));
+
+        final Graph graph = obj2.getTriples(subjects, PROPERTIES).collect(toModel()).getGraph();
+        final Node s = createGraphSubjectNode(obj2);
+        final ExtendedIterator<Triple> iter = graph.find(s, LAST_MODIFIED_DATE.asNode(), ANY);
+        assertTrue("Should have one lastModified triple", iter.hasNext());
```

Comment:
Maybe just `assertEquals` and `org.apache.jena.atlas.iterator.Iter.count(Iterator<T>)`?